nodejs/eks/cluster.ts

```diff
@@ -80,6 +80,29 @@ export interface CoreData {
     vpcCni?: VpcCni;
 }
 
+function translateAWSProviderToIAMAuth(clusterName: string, providerProfile?: string, providerAssumeRole?: { roleArn?: string }): pulumi.Output<any> {
+    const command = "aws-iam-authenticator";
+    const args = ["token", "-i", clusterName];
+    return pulumi.all([providerProfile, providerAssumeRole]).apply(([profile, assumeRole]) => {
+        const result: {[k: string]: any} = { args, command, apiVersion: "client.authentication.k8s.io/v1alpha1" };
```

Comment:
Perhaps instead of this - just define the interface we expect and then use that here?  A little heavier, but hopefully makes this easier to ensure is correct?

Reply:
Fixed